src/c_kzg_4844.c

```diff
@@ -229,6 +224,31 @@ static C_KZG_RET new_fr_array(fr_t **x, size_t n) {
     return c_kzg_calloc((void **)x, n, sizeof(fr_t));
 }
 
+/**
+ * Allocate memory for a polynomial.
+ *
+ * @remark Free the space later using free_poly().
+ *
+ * @param[out] out    Pointer to the poly_t struct
+ * @param[in]  length The length of the polynomial
+ */
+static C_KZG_RET new_poly(poly_t *out, uint64_t length) {
```

Comment:
I know that I was pressuring you to make a new type called `poly_t` instead of having it raw. But now that I see how it looks like, I think your initial intuition was correct.

My suggestion would be to switch to `fr_t *polynomial` and `uint8_t *blob` as it was in ef592fcd5de64c3623cac7bef3cd7e7ebaf8c2ba.

One thing that would be nice is to always be consistent about the naming of the above. So for example not `fr_t *p`, but instead always `fr_t *polynomial`. We have less type-safety this way, so let's use the variable names to our advantage.

Reply:
> I know that I was pressuring you to make a new type called `poly_t` instead of having it raw. But now that I see how it looks like, I think your initial intuition was correct.

Did you do `typedef fr_t *poly_t`?

I can't see the commit with `poly_t` anymore.

Reply:
For documentation purpose, the commit is https://github.com/ethereum/c-kzg-4844/commit/ff405cafe49a9bd5e108bf1a82ebe9c802753084